src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1446,6 +1455,28 @@ internal AnnotationModel AddAnnotation(string text, Guid id)
             return null;
         }
 
+        internal AnnotationModel AddAnnotation(string titleText, string text, Guid id)
+        {
+            var selectedNodes = this.Nodes == null ? null : this.Nodes.Where(s => s.IsSelected);
+            var selectedNotes = this.Notes == null ? null : this.Notes.Where(s => s.IsSelected);
+
+            if (!CheckIfModelExistsInSomeGroup(selectedNodes, selectedNotes))
+            {
+                var annotationModel = new AnnotationModel(selectedNodes, selectedNotes)
+                {
+                    GUID = id,
+                    AnnotationDescriptionText = titleText,
+                    AnnotationText = text
+                };
+                annotationModel.ModelBaseRequested += annotationModel_GetModelBase;
+                annotationModel.Disposed += (_) => annotationModel.ModelBaseRequested -= annotationModel_GetModelBase;
```

Comment:
Do we need to unsubscribe these somewhere?

Reply:
Should already be unsubscribed on Disposed?
`annotationModel.Disposed += (_) => annotationModel.ModelBaseRequested -= annotationModel_GetModelBase;`

Reply:
Yup, my bad missed this